File: cucumber_java/legacy_applications/01/src/test/java/legacy/Steps.java
/***
 * Excerpted from "The Cucumber for Java Book",
 * published by The Pragmatic Bookshelf.
 * Copyrights apply to this code. It may not be used to create training material, 
 * courses, books, articles, and the like. Contact us if you are in doubt.
 * We make no guarantees that this code is fit for any purpose. 
 * Visit http://www.pragmaticprogrammer.com/titles/srjcuc for more book information.
***/
package legacy;

import cucumber.api.java.en.*;

import org.junit.*;

public class Steps {
    @Given("^the price of a bottle of shampoo is \\$(\\d+)$")
    public void thePriceOfABottleOfShampooIs$(int itemPrice) throws Throwable {
    }

    @When("^I scan (\\d+) bottles of shampoo$")
    public void iScanBottlesOfShampoo(int itemCount) throws Throwable {
    }

    @Then("^the price should be \\$(\\d+)$")
    public void thePriceShouldBe$(int expectedTotal) throws Throwable {
        int priceCharged = 2; // Here we would call our legacy system
        Assert.assertEquals(expectedTotal, priceCharged);
    }
}

File: cucumber/legacy_applications/01/features/step_definitions/steps.rb
#---
# Excerpted from "The Cucumber Book, Second Edition",
# published by The Pragmatic Bookshelf.
# Copyrights apply to this code. It may not be used to create training material,
# courses, books, articles, and the like. Contact us if you are in doubt.
# We make no guarantees that this code is fit for any purpose.
# Visit http://www.pragmaticprogrammer.com/titles/hwcuc2 for more book information.
#---
Given /^the price of a bottle of shampoo is \$(\d+)\.(\d+)$/ do |arg1, arg2|
end

When /^I scan (\d+) bottles of shampoo$/ do |arg1|
end

Then /^the price should be \$(\d+)$/ do |arg1|
  expect(1.99).to eq 0
end
